subsys/bluetooth/host/keys.c

```diff
@@ -56,6 +56,21 @@ struct bt_keys *bt_keys_get_addr(u8_t id, const bt_addr_le_t *addr)
 	return NULL;
 }
 
+void bt_foreach_bond(u8_t id, void (*func)(const bt_addr_le_t *addr,
+					   void *user_data),
+		     void *user_data)
+{
+	int i;
+
+	for (i = 0; i < ARRAY_SIZE(key_pool); i++) {
+		struct bt_keys *keys = &key_pool[i];
+
+		if (keys->keys && keys->id == id) {
```

Comment:
We should probably skip BT_ADDR_LE_ANY and BT_ADDR_LE_NONE.

Reply:
A non-zero keys->keys should be a sufficient guarantee that keys->addr is valid.

Reply:
To elaborate: keys->keys is a bit-field of available keys for that entry (LTK, CSRK, IRK, etc). If any of the bits is set then the bt_keys entry is valid and the address is too.

Reply:
Ok then.